docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,149 @@ If you need to have external labels in your query results, this is currently not
+    
+    ```bash
+    gsutil -m cp -r gs://<THANOS-BUCKET>/* gs://<INTERMEDIATE-MIMIR-BUCKET>/
+    ```
+    
+    Once the copy process is done, inspect the blocks in the bucket to ensure it's valid from Thanos perspective.
+    
+    ```bash
+    thanos tools bucket inspect --objstore.config-file bucket.yaml
+    ```
+
+2) Remove the downsampled blocks
+
+   Mimir doesn't understand the downsampled blocks(ie. blocks with non-zero Resolution field in meta.json file) in Thanos. Therefore we need to remove the 5m and 1h downsampled blocks from this bucket.
+   
+   Mark the downsampled blocks for deletion.
+   
+   ```bash
+   thanos tools bucket retention --objstore.config-file bucket.yaml --retention.resolution-1h=2h --retention.resolution-5m=2h --retention.resolution-raw=0d
+   ```
+   Cleanup the blocks marked for deletion.
+   
+   ```bash
+   thanos tools bucket cleanup --objstore.config-file bucket.yaml --delete-delay=0
+   ```
+  
+3) Remove the duplicated blocks
+
+   If two replicas of Prometheus instancesare deployed for HA, then only upload the blocks from one of the replicas and drop the other.
+   
+   ```bash
+   thanos tools bucket inspect --objstore.config-file bucket.yaml --output=tsv | grep prometheus_replica=<PROMETHEUS-REPLICA-TO-DROP> | awk '{print $1}' | xargs -n 1 -P 1 thanos tools bucket mark --marker="deletion-mark.json" --objstore.config-file bucket.yaml --details="Removed as duplicate" --id
+   ```
+   
+   > **Note**: Replace **prometheus_replica** with the unique label that would differentiate prometheus replicas in your setup.
+
+   Again cleanup the duplicate blocks marked for deletion.
+  
+   ```bash
+   thanos tools bucket cleanup --objstore.config-file bucket.yaml --delete-delay=0
+   ```
+  
+   > **ProTip!**: If you want to visualize what exactly is happening in the blocks with respect to the source of blocks, labels, compaction levels, etc, you can use the below command to get the output as CSV and import it to a google sheet.
+   > ```bash 
+   > thanos tools bucket inspect --objstore.config-file bucket-prod.yaml --output=csv >> thanos-blocks.csv
+   > ```
+
+4) Relabel the blocks with external labels
+
+   Relabel the blocks with the required external labels. Create a rewrite config like the following:
+   
+   ```bash
+   # relabel-config.yaml
+   - action: replace
+     target_label: "<LABEL-KEY>"
+     replacement: "<LABEL-VALUE"
+   ```
+   
+   Perform the rewrite dry run to confirm all work well.
+   
+   ```bash
+   thanos tools bucket rewrite  --objstore.config-file bucket.yaml --rewrite.to-relabel-config-file relabel-config.yaml --dry-run --id
+   ```
+   
+   After confirming that the rewrite is working as expected in **--dry-run**, you can apply the changes with the **--no-dry-run** flag.
+   
+   ```bash
+   thanos tools bucket rewrite  --objstore.config-file bucket.yaml --rewrite.to-relabel-config-file relabel-config.yaml --no-dry-run --delete-blocks --id
```

Comment:
This needs to be run for all blocks in the bucket, right? `--id` parameter at the end is missing block ID. (Same comment for "dry run" above)

Reply:
My bad I missed adding the first part of the command. Since I am using xarg we don't have to pass the id specifically there it will be automatically passed. If I would have used the for loop then i need to value through a variable there.